### PR TITLE
Fix for conversion loading.

### LINKED ISSUE #721: automatic conversion of python lists in parameters does not work until java.util.Collection(?) is imported
Assume the following Java class:
```java
package jptest;
import java.util.Collection;

public class Test {
    public static void testMethod(Collection<String> input) {
        System.out.println(input);
    }
}
```
invoked by the following python code:
```python
import jpype

jpype.startJVM(
    jpype.getDefaultJVMPath(),
    classpath='/path/to/built/class,
    convertStrings=True
)

import jpype.imports
jpype.imports.registerDomain("jptest")

# from java.util import Collection

from jptest import Test

Test.testMethod(["a", "b"])
```

Expected result:
```
[a, b]
```

Actual result:
```
Traceback (most recent call last):
  File "D:/jpype-playground/test.py", line 16, in <module>
    Test.testMethod(["a", "b"])
TypeError: No matching overloads found for jptest.Test.testMethod(list), options are:
	public static void jptest.Test.testMethod(java.util.Collection)
```

Once `java.util.Collection`, `java.util.ArrayList` (or probably any other collection?) is imported, e.g. in the commented-out line, conversion works as expected.

JPype 0.7.3, Java 1.8 (x64) on Windows 10.

### PR DESCRIPTION
**Pull request by Thrameos**
This refreshes the Python type cache before matching arguments.  Without this user defined conversions may be missed.

Fixes #721

### CODE REVIEW

File: native/common/jp_classhints.cpp [hunk 1]
```diff
@@ -244,8 +244,13 @@ class JPHintsConversion : public JPConversion
 	virtual JPMatch::Type matches(JPMatch &match, JPClass *cls)
 	{
 		PyJPClassHints *pyhints = (PyJPClassHints*) cls->getHints();
+		// GCOVR_EXCL_START
+		if (pyhints == NULL)
+			PyJPClass_create(*match.frame, cls);
+		pyhints = (PyJPClassHints*) cls->getHints();
```
Comment (marscher) [inline native/common/jp_classhints.cpp:250]: this should go into the pyhints == NULL block, right?

File: test/harness/jpype/collection/CollectionTest.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package jpype.collection;
```
Comment (marscher) [inline test/harness/jpype/collection/CollectionTest.java:1]: is this file used elsewhere?

Comment (Thrameos) [reply test/harness/jpype/collection/CollectionTest.java:1]: This was a left over from the user report.  I think I should make a test that passes a Python list and make sure I get an array list out the other side.  Same with map

File: native/common/jp_classhints.cpp (hunk 1 above)
Comment (Thrameos) [reply native/common/jp_classhints.cpp:250]: Yeah it is not necessary to look it up twice.  Will fix.

**Pull request merged**